### DISCUSSION TITLE
next auth google login get custom token

### DISCUSSION THREAD

**Discussion by guitar9** [nextauthjs/next-auth#6322] [Help]
### Question 💬

I am using next js, next auth and a custom backend (flask), where I am using API Endpoint secured with jwt token. 
In the next js frontend I want to access this jwt token with the useSession hook to make requests to this API Endpoints. With the CredentialsProvider everything works fine. I was able to register and login a user and can get access the jwt token. When I am using useSession with credentials login, I have the data in the following format, which is what i want (token.sub is my jwt access token) :

    {
        "data": {
            "session": {
                "user": {},
                "expires": "2023-02-06T21:35:50.296Z"
            },
            "token": {
                "sub": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJmcmVzaCI6ZmFsc2UsImlhdCI6MTY3MzEyNzM1MCwianRpIjoiZDJmODIwZjAtOGYxOS00Y2Q5LTk0Yxxxxxxx",
                "iat": 1673127350,
                "exp": 1675719350,
                "jti": "53b9fcaa-3aa9-4567-9381-341231144e72"
            }
        },
        "status": "authenticated"
    }

however when I use the google login I am getting this format: 
All I want is to get the jwt access token from my backend. 

    {
        "data": {
            "session": {
                "user": {
                    "name": "google name",
                    "email": "google email",
                    "image": "google pic"
                },
                "expires": "2023-02-06T21:24:09.905Z"
            },
            "token": {
                "name": "google user name",
                "email": "google email",
                "picture": "google pic",
                "sub": "115092xxxxxxx",
                "iat": 16731xxxx,
                "exp": 1675xxxx6,
                "jti": "1fc246fa-xxxxxx"
            }
        },
        "status": "authenticated"
    }

This is the code for next-auth. To get the access with google login I added the signIn callback and add there the request to my backend to get the jwt access token. 

    import NextAuth, { NextAuthOptions } from "next-auth";
    import CredentialsProvider from "next-auth/providers/credentials";
    import GoogleProvider from "next-auth/providers/google"
    import API from "../../../API";
    import axiosHandler from "../../../axios-handler";
    
    
    const authOptions = {
      session: {
        strategy: "jwt",
      },
      providers: [
        CredentialsProvider({
          type: "credentials",
          credentials: {},
          authorize: async (credentials, req) => {
            const loginResponse = await axiosHandler.post(credentials.isRegister? API.REGISTER :API.LOGIN, credentials)
            if(loginResponse.data){
              return {
                id: loginResponse.data.token,
                token: loginResponse.data.token,            
              }
            }
            return null
          },
        }),
        GoogleProvider({
          clientId: process.env.GOOGLE_CLIENT_ID,
          clientSecret: process.env.GOOGLE_CLIENT_SECRET,
        })
      
      ],
      pages: {
        signIn: "/auth/signin",
      },
      callbacks: {
        async session({ session, user, token }) {
          return {session, user, token}
        },
    
        async jwt({ token, user, account, profile, isNewUser }) {
          return token
        },
        async signIn({user, account}){
          if(account?.provider ==="google"){
            const googleAuthData = {
              name: user.name,
              email: user.email,
              image: user.image,
              authProvider: 'google',
              password: ''
          }
          const loginResponse = await axiosHandler.post(API.LOGIN, 
            {
              name: googleAuthData.name, 
              email: googleAuthData.email, 
              secretKey:"xxx"})
          if(loginResponse.data){
            account.test =  loginResponse.data.token
            user.test = loginResponse.data.token
            return true
          }
          }
          return true
          
        }
      },
    };
    
    export default NextAuth(authOptions);

### How to reproduce ☕️

See above

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
https://next-auth.js.org/getting-started/example#extensibility

**Comment by msobkyy** [comment]
did u find a solution?

**Comment by Enkratia** [comment]
Does anyone know why `jwt` callback always return undefined for `account`, `profile`, `session`? 
(`token` and `user` are working).
